### ISSUE TITLE
Write article about docker-machine

### ISSUE THREAD

**Issue by hatchan** [wercker/docs#318]
Currently there are no guidelines to using wercker together with docker-machine. Currently we have these instructions: https://github.com/wercker/support/issues/119#issuecomment-139503221

These should be expanded and fit into a devcenter article.

**Comment by austenLacy** [comment]
hey @hatchan I'm actually trying out wercker using docker-machine instead of boot2docker and am having some issues with running `wercker dev --publish 8080` with my Golang project. However, I
**think** was able to get things everything involving docker-machine over boot2docker working so if you want I can add some stuff to the docs?

**Comment by hatchan** [comment]
@austenLacy I was probably gonna do this on Friday. So if you want to create a PR with an atricle, or just give the feedback in this item, that would be appreciated :smiley_cat:

**Comment by austenLacy** [comment]
---
tags: cli
---

## Requirements

In order to make use of all of the features within the CLI you will need
a working Docker environment. If you are running Mac OSX or Windows you can install Docker and Docker-Machine using [Docker Toolbox](https://www.docker.com/docker-toolbox). Docker Toolbox is an installer to quickly and easily install and setup a Docker environment on your computer. Toolbox installs the Docker Client, Docker Machine, Docker Compose, Docker Kitematic and VirtualBox.

On linux it is easy to install Docker depending on your distribution and package manager. See Docker's [installation instructions](https://docs.docker.com/installation/#installation) for more information and for Docker Machine see [here](https://docs.docker.com/machine/install-machine/).

If you haven't already, download [Docker Toolbox](https://www.docker.com/docker-toolbox).

### OSX Docker Machine quick start

Essentially Docker Machine creates a host on an instance of Virtual Box that you can then use with the Docker client to build, run, and monitor your Docker containers.

From the official [Docker documentation](https://docs.docker.com/machine/),
>Machine lets you create Docker hosts on your computer, on cloud providers, and inside your own data center. It automatically creates hosts, installs `Docker` on them, then configures the docker client to talk to them. A “machine” is the combination of a Docker host and a configured client.

To confirm there weren't any hiccups try entering the command `docker-machine ls` in a shell session. If everything went well and assuming this is the first time you are running the command, you should see something like this

```no-highlight
$ docker-machine ls
NAME   ACTIVE   DRIVER   STATE   URL
```

Notice there currently aren't any machines listed, which makes sense since you haven't created any yet. 

To create a machine, we run the `docker-machine create` command, passing the string `virtualbox` to the `--driver` flag. The final argument we pass is the name of the machine - in this case, we will name our machine `dev`, but you can name it whatever you want.

You should see something like
```no-highlight
$ docker-machine create --driver virtualbox dev
Running pre-create checks...
Creating machine...
Waiting for machine to be running, this may take a few minutes...
Machine is running, waiting for SSH to be available...
Detecting operating system of created instance...
Provisioning created instance...
Copying certs to the local machine directory...
Copying certs to the remote machine...
Setting Docker configuration on the remote daemon...
To see how to connect Docker to this machine, run: docker-machine env dev
```

Now if we run `docker-machine ls` again we should see our newly created `dev` machine.

```no-highlight
$ docker-machine ls
NAME   ACTIVE   DRIVER       STATE     URL                         SWARM
dev    -        virtualbox   Running   tcp://192.168.99.100:2376
```

That's all there is to it to create a machine and get it running! For more info on `docker-machine` and its features see [here](https://docs.docker.com/machine/get-started/).

Now we need to tell `Docker` to be able to talk to your previously created `dev` machine. We do that with the `docker-machine env` command.

In order to see the Docker certificates and environment variables you can run the `docker-machine env YOUR-MACHINE-NAME-HERE` command.

It should look something like this

```no-highlight
$ docker-machine env dev
export DOCKER_TLS_VERIFY="1"
export DOCKER_HOST="tcp://192.168.99.101:2376"
export DOCKER_CERT_PATH="$HOME/.docker/machine/machines/dev"
export DOCKER_MACHINE_NAME="dev"
# Run this command to configure your shell:
# eval "$(docker-machine env dev)"
```

In order to configure our shell with the appropriate Docker environment variables and certificates
we need to run the `eval "$(docker-machine env YOUR-MACHINE-NAME-HERE)"` command in our shell. If everything was successful you shouldn't see any output from that command. 

> Note that this only activates Docker in the **current** shell session

Since this only works for the current shell session and to avoid having to remember to run the `eval "$(docker-machine env YOUR-MACHINE-NAME-HERE)"` command each time you open a new shell session you might consider adding the `exports` from the `docker-machine env dev` command to your `.profile`, `.bashrc`, `.zshrc`, or other shell configuration file.

And you're done! You should now have Docker Client, Docker Machine, Docker Compose, Docker Kitematic and VirtualBox installed, at least one Docker Machine created and running, and configured Docker to communicate to that machine.

**Comment by austenLacy** [comment]
@hatchan let me know if something like that will work, thanks!

**Comment by hatchan** [comment]
@austenLacy Awesome, very nice. Will look at it this Friday :+1:

**Comment by austenLacy** [comment]
@hatchan no problem, glad to help!

**Comment by KorvinSzanto** [comment]
@austenlacy These are the same steps I took to arrive at https://github.com/wercker/support/issues/57

**Comment by faiq** [comment]
@austenLacy i read over what you wrote, and it looks good to me. thanks for putting it together. i was able to get wercker up and running with docker machine and what you wrote here pretty accurately describes the steps i went through to set it up. 

right now the requirements section of our docs describes on building docker using boot2docker, and i think it would make sense to replace it with what you wrote here. since you already put in the effort of writing a guide on building docker using docker-machine, do you want to open up a pull request replacing our current requirements section with what you wrote here? it would make more sense to review and comment on what you wrote there. if you don't want to, i can totally put it together for you :smile:

thanks again!

**Comment by austenLacy** [comment]
@faiq no problem. here you go --> https://github.com/wercker/docs/pull/328

**Comment by faiq** [comment]
@austenLacy thanks so much! i'll look it over and comment there!

**Comment by faiq** [comment]
fixed in #328 closing
